File: demo.in-jpaentity-med-validation/src/main/kotlin/com/example/demo/user_interface/UniqueSeteNummer.java

package com.example.demo.user_interface;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Constraint(validatedBy = UniqueSeteNummerConstraintValidator.class)
@Retention(RetentionPolicy.RUNTIME)
public @interface UniqueSeteNummer {
    String message() default "Setene må ha unike nummer i en kupe";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: demo.in-jpaentity-med-validation/src/main/kotlin/com/example/demo/user_interface/constraints/UniqueSeteNummer.kt
package com.example.demo.user_interface.constraints

import javax.validation.Constraint
import kotlin.reflect.KClass

@Constraint(validatedBy = [UniqueSeteNummerConstraintValidator::class])
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class UniqueSeteNummer(
        @Suppress("unused") val message: String = "Setene må ha unike nummer i en kupe",
        @Suppress("unused") val groups: Array<KClass<out Any>> = [],
        @Suppress("unused") val payload: Array<KClass<out Any>> = []
)
